src/commonMain/kotlin/uuid.kt

```diff
@@ -138,6 +140,11 @@ public class Uuid(val uuid: ByteArray) {
          * @param from The String, from which to deserialize the UUID
          * @return a UUID, if the string is a valid UUID string
          */
+        @Deprecated(
+            message = "Use Uuid.fromString() instead.",
+            replaceWith = ReplaceWith("Uuid.fromString(from)"),
+            level = WARNING
+        )
         fun parse(from: String): Uuid? {
```

Comment:
How about we update this one to do:
```kotlin
    try {
        return UUID.fromString(from)
    } catch (_: Throwable) {
        return nil
    }
```

Reply:
I actually considered doing that first but catching an exception is expensive (atleast on the JVM). I don't think penalizing old projects would be a good thing. Thoughts?

Reply:
In that case, the best way might just be to make a private method that returns a private sealed class. The sealed class would be like a Result type whose possible values are the value Uuid or an error to throw. In the new public method, we call the private method and throw the exception. In the old method, we return nil. Does that make sense? It's a bit more code, but it's probably the most accurate and allows deferring surfacing the error to the caller

Reply:
The other thing I'm thinking about:

1. If it's only the throwing/catching part that's slow, maybe we don't worry about it because that's "exceptional" case?
2. This only applies to the deprecated method. The solution for users is then to just apply the deprecation inspection/fix

So maybe we don't worry about it?

Reply:
> If it's only the throwing/catching part that's slow, maybe we don't worry about it because that's "exceptional" case?

That… is a good point. Passing in invalid strings is an error anyway. I'll go ahead with catching exceptions then?

Reply:
Sounds good! So no more changes right? Good as is?

Reply:
Oh never mind I see the consolidation still has to happen. Will wait for that :)

Reply:
Aaand done. Up for a final review!